lib/evse_security/crypto/openssl/openssl_supplier.cpp

```diff
@@ -0,0 +1,557 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+#include <evse_security/crypto/openssl/openssl_supplier.hpp>
+#include <evse_security/detail/openssl/openssl_types.hpp>
+#include <evse_security/utils/evse_filesystem.hpp>
+
+#include <everest/logging.hpp>
+
+#include <chrono>
+
+#include <openssl/bio.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/pem.h>
+#include <openssl/sha.h>
+#include <openssl/x509v3.h>
+
+namespace evse_security {
+
+struct X509HandleOpenSSL : public X509Handle {
+    X509HandleOpenSSL(X509* certificate) : x509(certificate) {
+    }
+
+    virtual void* get_raw_handle() override {
+        return x509.get();
+    }
+
+    X509* get() {
+        return x509.get();
+    }
+
+private:
+    X509_ptr x509;
+};
```

Comment:
Is the intention that `X509HandleOpenSSL` should only be accessible within this file? If not, I would move this to the header.

Reply:
Yes, it is a private extension of the generic handle, only the crypto supplier should see this, nobody else should have access to the X509* internal handle. For other crypto suppliers like WolfSSL it will be WOLFSSL_X509* or mbedtls_x509_crt* mbedtls. Those handles must never be seen from the exterior.